src/gc/proxy.d

```diff
@@ -68,132 +103,159 @@ extern (C)
         // NOTE: Due to popular demand, this has been re-enabled.  It still has
         //       the problems mentioned above though, so I guess we'll see.
 
-        instance.collectNoStack(); // not really a 'collect all' -- still scans
-                                    // static data area, roots, and ranges.
+        if (instance !is null)
```

Comment:
I'm not familiar with who calls `gc_term`.  Would it be better to make this synchronized like you have done in `gc_init`?  Or is this only going to be called by a single thread?

Reply:
```gc_term``` is called in ```dmain2.d->rt_term``` after ```thread_joinAll```.
https://github.com/dlang/druntime/blob/master/src/rt/dmain2.d#L223